gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-client/src/test/java/org/apache/gobblin/service/FlowConfigV2Test.java

```diff
@@ -262,7 +265,7 @@ public void testGroupRequesterRejected() throws Exception {
     }
   }
 
-  @Test
+  @Test(dependsOnMethods = {"testGroupRequesterRejected", "testGroupRequesterAllowed", "testGroupUpdateRejected", "testRequesterUpdate"})
```

Comment:
Do we have a way to run this without dependencies? Test dependencies bring a couple of problems:

- We can't run all tests in parallel, which affects build time
- To understand what goes on in the test, developer also need to read all dependent tests
- People that add new tests will likely be unaware about dependencies, and will just add their test to the bottom. This can break old tests in unexpected ways.

Usually, it's good to have tests isolated from each other. If you need a common setup logic, it can be moved to a separate method and called from all needed tests.

Reply:
I couldn't think of a good way since it is modifying files that were used by other tests in the same class (this could have been a problem even before this PR).

I decided to refactor and move this test to a separate class. I'm not sure why it was even in this class in the first place, since it seems to be testing that `LocalGroupOwnershipService` is able to pick up changes to the file, which is unrelated to flowConfig client. (And in this class there is already tests that group ownership is working while calling flowConfig client).